x-pack/plugin/eql/src/test/java/org/elasticsearch/xpack/eql/action/EqlSpecLoader.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.action;
+
+import io.netty.util.internal.StringUtil;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EqlSpecLoader {
+    public static List<EqlSpec> load(String path, boolean supported) throws Exception {
+        try (InputStream is = EqlSpecLoader.class.getResourceAsStream(path)) {
+            return readFromStream(is, supported);
+        }
+    }
+
+    private static int[] readExpectedEventIds(String line, BufferedReader reader) throws Exception {
+        String arr[] = readArray(line);
+        int ids[] = new int[arr.length];
+
+        for (int i = 0; i < arr.length; i++) {
+            ids[i] = Integer.parseInt(arr[i]);
+        }
+        return ids;
+    }
+
+    private static String[] readTags(String line) throws Exception {
+        String arr[] = readArray(line);
+        for (int i = 0; i < arr.length; i++) {
+            String s = arr[i];
+            if (s.startsWith("\"") || s.startsWith("'")) {
+                s = s.substring(1);
+            }
+            if (s.endsWith("\"") || s.endsWith("'")) {
+                s = s.substring(0, s.length() - 1);
+            }
+            arr[i] = s.trim();
+        }
+        return arr;
+    }
+
+    private static String readValueLine(String line) throws Exception {
+        int idx = line.indexOf("=");
+        if (idx == -1) {
+            throw new IllegalArgumentException("Invalid string value: " + line);
+        }
+        return line.substring(idx + 1).trim();
+    }
+
+    private static String[] readArray(String line) throws Exception {
+        line = readValueLine(line);
+        if (!line.startsWith("[") && !line.endsWith("]")) {
+            throw new IllegalArgumentException("Invalid array string value: " + line);
+        }
+        String arr[] = line.substring(1, line.length() - 1).split(",");
+
+        ArrayList<String> res = new ArrayList<>();
+        for (String s : arr) {
+            s = s.trim();
+            if (!s.isEmpty()) {
+                res.add(s);
+            }
+        }
+
+        return res.toArray(new String[res.size()]);
+    }
+
+    private static String readString(String line, BufferedReader reader) throws Exception {
+        line = readValueLine(line);
+        String delim = "";
```

Comment:
I don't think this initialization is necessary. A simple `String delim;` should be enough.

Reply:
IDE was complaining on unitialized var delim otherwise